src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,165 @@
+bounds.-/
+theorem generate_measurable_subset_rec (s : set (set α)) ⦃t : set α⦄
+  (ht : generate_measurable s t) :
+  t ∈ ⋃ i, generate_measurable_rec s i :=
+begin
+  haveI : nonempty ω₁, by simp [← mk_ne_zero_iff, ne_of_gt, (aleph 1).mk_ord_out, aleph_pos 1],
+  inhabit ω₁,
+  induction ht with u hu u hu IH f hf IH,
+  { refine mem_Union.2 ⟨default, _⟩,
+    rw generate_measurable_rec,
+    simp only [hu, mem_union_eq, true_or] },
+  { refine mem_Union.2 ⟨default, _⟩,
+    rw generate_measurable_rec,
+    simp only [union_singleton, mem_union_eq, mem_insert_iff, eq_self_iff_true, true_or] },
+  { rcases mem_Union.1 IH with ⟨i, hi⟩,
+    obtain ⟨j, hj⟩ : ∃ j, i <₁ j := ordinal.has_succ_of_is_limit
+      (by { rw ordinal.type_out, exact ord_aleph_is_limit 1 }) _,
+    apply mem_Union.2 ⟨j, _⟩,
+    rw generate_measurable_rec,
+    have : ∃ a, (a <₁ j) ∧ u ∈ generate_measurable_rec s a := ⟨i, hj, hi⟩,
+    simp [this] },
+  { have : ∀ n, ∃ i, f n ∈ generate_measurable_rec s i := λ n, by simpa using IH n,
+    choose I hI using this,
+    obtain ⟨j, hj⟩ : ∃ j, ∀ k, k ∈ range I → (k <₁ j),
+    { apply ordinal.lt_cof_type,
+      simp only [is_regular_aleph_one.2, mk_singleton, ordinal.type_out],
+      have : #(range I) = lift.{0} (#(range I)), by simp only [lift_uzero],
+      rw this,
+      apply mk_range_le_lift.trans_lt _,
+      simp [omega_lt_aleph_one] },
+    apply mem_Union.2 ⟨j, _⟩,
+    rw generate_measurable_rec,
+    have : ∃ (g : ℕ → (↥⋃ (i : {i // i <₁ j}), generate_measurable_rec s i.1)),
+      (⋃ (n : ℕ), ↑(g n)) = (⋃ n, f n),
+    { refine ⟨λ n, ⟨f n, _⟩, rfl⟩,
+      exact mem_Union.2 ⟨⟨I n, hj (I n) (mem_range_self _)⟩, hI n⟩ },
+    simp [this] }
+end
+
+/-- If a sigma-algebra is generated by a set of sets `s`, then the sigma
+algebra has cardinality at most `(max (#s) 2) ^ ω`. -/
+theorem cardinal_generate_measurable_le (s : set (set α)) :
+  #{t | generate_measurable s t} ≤ (max (#s) 2) ^ omega.{u} :=
+(mk_subtype_le_of_subset (generate_measurable_subset_rec s)).trans
+  (cardinal_Union_generate_measurable_rec_le s)
+
+/-- If a sigma-algebra is generated by a set of sets `s`, then the sigma
+algebra has cardinality at most `(max (#s) 2) ^ ω`. -/
+theorem cardinal_measurable_set_le (s : set (set α)) :
+  #{t | @measurable_set α (generate_from s) t} ≤ (max (#s) 2) ^ omega.{u} :=
+cardinal_generate_measurable_le s
+
+/-- If a sigma-algebra is generated by a set of sets `s` with cardinality at most the continuum,
+then the sigma algebra has the same cardinality bound. -/
+theorem cardinal_generate_measurable_le_continuum {s : set (set α)} (hs : #s ≤ 𝔠) :
+  #{t | generate_measurable s t} ≤ 𝔠 :=
+calc
+#{t | generate_measurable s t} ≤ (max (#s) 2) ^ omega.{u} : cardinal_generate_measurable_le s
+... ≤ 𝔠 ^ omega.{u} :
+begin
+  apply power_le_power_right,
+  convert max_le hs (nat_lt_continuum 2).le,
+  simp,
+end
+... = 𝔠 : continuum_power_omega
```

Comment:
```suggestion
(cardinal_generate_measurable_le s).trans begin
  rw ←continuum_power_omega,
  exact_mod_cast power_le_power_right (max_le hs (nat_lt_continuum 2).le)
end
```

Reply:
Golfing should not be a goal in itself: preserving readability for nontrivial stuff is IMHO much more important.